src/Pickles/Pickles.TestFrameworks.UnitTests/StandardTestSuite.cs

```diff
@@ -164,6 +164,31 @@ public void ThenCanReadNotFoundFeatureCorrectly()
 
             Check.That(result).IsEqualTo(TestResult.Inconclusive);
         }
+    
+
+        public void ThenCanReadFeatureWithMatchingNameAndNoCategoriesCorrectly()
+        {
+            var results = ParseResultsFile();
+            var result = results.GetFeatureResult(new Feature { Name = "Adding several numbers" });
+
+            Check.That(result).IsEqualTo(TestResult.Failed);
+        }
+
+        public void ThenCanReadFeatureWithMatchingNameAndCategoryCorrectly()
+        {
+            var results = ParseResultsFile();
+            var result = results.GetFeatureResult(new Feature { Name = "Adding several numbers", Tags = { "@tag2" }});
+
+            Check.That(result).IsEqualTo(TestResult.Passed);
+        }
+
+        public void ThenCanReadFeatureWithMatchingNameAndUnknownCategoryCorrectly()
```

Comment:
Note that in this case, we expect to match the first xml element encountered that has the same description as the feature. This matches the previous behavior, so that we don't break the tool for users who rely on nunit results that don't have a Category element for each tag present in the gherkin.